src/main/java/com/fauna/client/RequestBuilder.java

```diff
@@ -95,6 +101,21 @@ public HttpRequest buildRequest(Query fql, QueryOptions options, CodecProvider p
         }
     }
 
+    public HttpRequest buildStreamRequest(StreamRequest req, CodecProvider provider) {
+        HttpRequest.Builder builder = baseRequestBuilder.copy();
+        Codec<StreamRequest> codec = provider.get(StreamRequest.class);
+        try {
+            UTF8FaunaGenerator gen = new UTF8FaunaGenerator();
+            codec.encode(gen, req);
+            gen.writeEndObject();
```

Comment:
Seems like something may have gone wrong if we're using a codec but still need to write an end object

Reply:
I wasn't sure if this was user error or not, will fix!